### DISCUSSION TITLE
Question about the nsc.FVT table

### DISCUSSION THREAD

**Discussion by fortrans** [GarnesDoug/FVT-Baylor-University#5] [Q&A]
Thanks for sharing your scripts!  Is the nsc.FVT table in the merge scripts (NSC2223 tSQL_Merge.sql and NSC2324 tSQL_Merge.sql) created from the FVT_Shared.sql script?  Seems like it must be, but then some of the field names are wrong (i.e. f.RecordType should be f.NSCRecordType) and some do not exist in nsc.FVT table (i.e. f.INVALIDFLAG).  Am I doing something wrong or do I just need to modify those fields?

**Comment by fmc5-git** [comment]
I believe  that you are talking about the temp table that they use. I think that you would need to work with your systems department to allow the nsc list files to be uploaded to the temp table. 

For testing purposes, I created a manual temp table with a few id's as well as the necessary TA and AA codes.

I started my script with something like this.

WITH TermAndAidYear AS (
  -- Hardcoded values for FA (Fall) and SPR (Spring) aid periods with specific term codes and aid year 2324
  SELECT
    '2223' AS AidYear,    -- Fix this for the correct format 20222023
    '202310' AS TermCode -- FA term
  UNION
  ALL
  SELECT
    '2223',    -- Fix this for the correct format 20222023
    '202230' -- SPR term
),
em_tmp AS (
  SELECT
    *
  FROM
    (
      VALUES
        ('TA', '123456789', 'G00123456', '110701', '2223'),  -- Fix this for the correct format 20222023
        ('AA', '123456789', 'G00123456', '110701', '2223')  -- Fix this for the correct format 20222023
    ) AS em_tmp (
      NSCRecordType,
      NSCSSN,
      NSCStudentID,
      NSCCIPCode,
      NSCAidYear
    )
),

**Reply by fortrans** [reply]
I was asking if the nsc.FVT table referenced in the merge scripts (NSC2223 tSQL_Merge.sql and NSC2324 tSQL_Merge.sql) was created using FVT_Shared.sql.

**Reply by fmc5-git** [reply]
You should get each of those files from NSLDS. You may need to reach out to NSLDS and find out if it is sent to a different TG/SAIG
mailbox.

**Comment by fortrans** [comment]
Thanks Matt.  Also did you have any trouble with the getStuRec section of FVT Shared.sql, specifically the join on major (see below)?  Because it's an inner join we are not getting results for students who's major doesn' t match what's in the NSLDS file due to major changes.

INNER JOIN stvmajr
     ON stvmajr.stvmajr_cipc_code= NSCList.NSCCIPCode
    AND stvmajr.stvmajr_code     = sgbstdn1.sgbstdn_majr_code_1
